matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncTokenStore.kt

```diff
@@ -25,13 +25,18 @@ import javax.inject.Inject
 internal class SyncTokenStore @Inject constructor(@SessionDatabase private val monarchy: Monarchy) {
 
     fun getLastToken(): String? {
-        return Realm.getInstance(monarchy.realmConfiguration).use {
+        val token = Realm.getInstance(monarchy.realmConfiguration).use {
+            // Makes sure realm is up-to-date as it's used for querying internally on non looper thread.
+            it.refresh()
             it.where(SyncEntity::class.java).findFirst()?.nextBatch
         }
+        return token
     }
 
     fun saveToken(realm: Realm, token: String?) {
-        val sync = SyncEntity(token)
-        realm.insertOrUpdate(sync)
+        if (realm.isInTransaction) {
```

Comment:
it should always be the case, no? Else shouldn't we do something like crashing the app for instance?

Reply:
oh yes sorry, miss from testing